.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Lint Go Code
+
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.19.0
```

Comment:
The project uses go version `1.21`. You can check this for example [here](https://github.com/keyval-dev/odigos/blob/78631271fd3123f01a78f5585cd503b3ea438a1a/.github/workflows/main.yml#L23) and [here](https://github.com/keyval-dev/odigos/blob/78631271fd3123f01a78f5585cd503b3ea438a1a/go.work#L1)